### ISSUE TITLE
Fastlane doesn't work for F-Droid

### ISSUE THREAD

**Issue by linsui** [johannesjo/super-productivity#3597]
F-Droid can only find the fastlane structure in root path. After the merging it's in `android` now. Could you please move it to root path? Thanks!

**Comment by johannesjo** [comment]
Thanks for opening this up! I'll investigate!

**Comment by johannesjo** [comment]
You seem to have more knowledge about the subject than I do. The android code now resides in the `android` subfolder. Can I just move the fastlane folder to the root folder anyway?

**Comment by linsui** [comment]
Yep, you can just move the fastlane to root folder. A simple symlink also works. :)

**Comment by johannesjo** [comment]
Thank you!!

**Comment by linsui** [comment]
Thanks! Will the android app release new versions along with the whole project? Do we need to pick special tags?

**Comment by johannesjo** [comment]
> Thanks! Will the android app release new versions along with the whole project? Do we need to pick special tags?

Yes! That is the plan. Not sure about the tags yet tbh. :)

Automated builds are not yet set up. Do you have any experience with that by chance?

**Comment by linsui** [comment]
If android release uses the same tag then we don't need to do anything. If you want F-Droid to pick any special tag, e.g. xxx-android, just ping me. :)

**Comment by johannesjo** [comment]
@linsui (and also @jiongxuan) I have a question regarding versioning and I thought you might be able to help me. For desktop and web we are currently on the version 10.0.11 for android we are on `versionCode` 30 and `versionName` "30.10.0.11". For desktop and web I just need to run `npm version patch` and then the `version` script in the package.json (and then github actions) takes care of everything else. How would you deal with android? I guess that we would need to extend the `version` script in the package.json with a script that upgrades the version inside the `build.gradle` file by upgrading `versionCode` by one regardless of the type of release. How would you approach this? Is there a ready standard solution for it?

**Comment by linsui** [comment]
It works for F-Droid as long as the version code is increased for every version. You can increase it by one. You can also convert the version name to version code, e.g. 3010011.

**Comment by jiongxuan** [comment]
@johannesjo @linsui, thanks for the suggestion!

The reason we initially went with “30.10.0.11” for Android was because the previous version was “23.0”, so aligning with the main version would appear as a “downgrade” to users. We also wanted to highlight this as a significant update, which is why we chose the 30.x.x.x format.

I agree with your approach and am considering converting it to a “30100011” format (with an extra zero for each segment, like 30.10.00.11). This would make the versioning easier to maintain. As long as there are no major changes, we could keep “30” consistent and just increment the patch numbers as needed. 

What do you both think of this approach?

**Comment by linsui** [comment]
LGTM!

**Comment by johannesjo** [comment]
> What do you both think of this approach?

Generally I'd prefer something like 100011 (major_minor_patch each with 2 digits), since that's already enough numbers and the 30 is just a random number. Would there be any other negative drawbacks apart from a little confusion that the new version number is below the current?

**Comment by jiongxuan** [comment]
@johannesjo  That sounds great! Just to clarify, are you suggesting this change for both versionName and versionCode, or only for versionCode? If it’s only versionCode, it shouldn’t be an issue since users won’t see it. However, if it’s also for versionName, there could be a perception of a “downgrade.”

I’ve actually been considering aligning versionName with the main version too. Even if it might cause a bit of confusion for users initially, it could simplify things overall. 

What do you think?

**Comment by johannesjo** [comment]
>  if it’s also for versionName, there could be a perception of a “downgrade.”

I think this is OK since I assume that users usually don't pay too much attention to this anyway. I personally usually don't for mobile apps :) Lets keep things simple for us working on the project :)

**Comment by linsui** [comment]
Yes, thanks!
